src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.cs

```diff
@@ -65,8 +65,8 @@ internal static partial class ConflictResolver
                         callbackTarget: null,
                         cancellationToken).ConfigureAwait(false);
 
-                    if (result.HasValue && result.Value != null)
-                        return await result.Value.RehydrateAsync(solution, cancellationToken).ConfigureAwait(false);
+                    if (result != null)
+                        return await result.RehydrateAsync(solution, cancellationToken).ConfigureAwait(false);
                 }
             }
```

Comment:
you're right.  this code is suspect.   i'll look into thsi once your PR goes in.